src/main/java/com/box/sdk/BoxEvent.java

```diff
@@ -167,8 +177,16 @@ void parseJsonMember(JsonObject.Member member) {
                 }
             }
 
-            if (this.type == null) {
+            for (EventType t : EventType.values()) {
+                if (t.name().equals(stringValue)) {
+                    this.eventType = t;
+                    break;
+                }
+            }
```

Comment:
I think using `valueOf` is not the best approach here. If we don't find the event in our `EventType` enum we return `EventType.UNKNOWN`. Controlling flow of the application in catch clauses is often considered as anti-pattern. Instead, I reimplemented `valueOf` that will return `null` instead of throwing an exception if value is not found. It's still O(1) like the original `valueOf`.

Reply:
Yes, I was thinking that the exception would happen so extremely rarely, that it would be acceptable, but actually, that may not be true for old versions of the SDK that are still in use, with an out-of-date EventType enum.